backend/src/serverless/microservices/nodejs/integration-data-checker/integrationDataCheckerWorker.ts

```diff
@@ -0,0 +1,122 @@
+ * @param integration The integration to check
+ * @param userContext User context
+ */
+async function checkIntegrationForAllSettings(integration, userContext: IRepositoryOptions) {
+  const activityService = new ActivityService(userContext)
+  for (const settings of integrationDataCheckerSettings) {
+    // This is moment() - the time. For example, moment().subtract(1, 'hour') is 1 hour ago.
+    const timestampSinceLastData = generateDate(settings.timeSinceLastData)
+
+    if (shouldCheckThisIntegration(integration, settings, timestampSinceLastData)) {
+      if (
+        !(settings.type === IntegrationDataCheckerSettingsType.PLATFORM_SPECIFIC) ||
+        settings.activityPlatformsAndType?.platforms.includes(integration.platform)
+      ) {
+        const activityCount = (
+          await activityService.findAndCountAll({
+            filter: {
+              platform: integration.platform,
+              createdAt: {
+                gte: timestampSinceLastData,
+              },
+              ...(settings.type === IntegrationDataCheckerSettingsType.PLATFORM_SPECIFIC && {
+                type: settings.activityPlatformsAndType.type,
+              }),
+            },
+            limit: 1,
+          })
+        ).count
+
+        if (!activityCount) {
+          await changeStatusAction(settings, integration, userContext)
+          await sendSlackAlertAction(settings, integration, userContext)
+          break
+        }
+      }
+    }
+  }
+}
+
+function shouldCheckThisIntegration(
+  integration: any,
+  settings: IntegrationDataCheckerSettings,
+  timestampSinceLastData: moment.Moment,
+) {
+  // We always should be only checking integrations that have been created before the time we want to check.
+  // Otherwise, it is never relevant.
+  if (integration.createdAt < timestampSinceLastData) {
+    // Either we do not care about it being only new integrations,
+    // or the integration's createdAt is before the time we want to check + the reset frequency.
+    return (
+      !settings.onlyNewIntegrations ||
+      integration.createdAt > timestampSinceLastData.subtract(1, 'hour')
+    )
+  }
+  return false
+}
+
+async function changeStatusAction(
+  settings: IntegrationDataCheckerSettings,
+  integration,
+  userContext: IRepositoryOptions,
+) {
+  if (settings.actions.changeStatus) {
+    const integrationService = new IntegrationService(userContext)
+    await integrationService.update(integration.id, {
+      status: 'no-data',
+    })
+  }
+}
+
+async function sendSlackAlertAction(
+  settings: IntegrationDataCheckerSettings,
+  integration,
+  userContext: IRepositoryOptions,
+) {
+  return sendSlackAlert(SlackAlertTypes.DATA_CHECKER, integration, userContext, log, settings)
+}
+
+function generateDate(timeframe) {
+  const now = moment()
+  if (timeframe.includes('hour')) {
+    const hours = parseInt(timeframe, 10)
+    return now.subtract(hours, 'hours')
```

Comment:
NIT:
I was confused here because we're sending an example string '2 hours' into `parseInt`. 
I didn't know parseInt was ignoring all the characters after whitespace. Can we split this string with whitespace and send/check split values for clarity?